src/brod_client.erl

```diff
@@ -154,38 +199,59 @@ get_producer(ClientId, Topic, Partition) when is_atom(ClientId) ->
     {error, client_down}
   end.
 
--spec find_producer(client_id(), topic(), partition()) ->
+-spec find_producer(client(), topic(), partition()) ->
                        {ok, pid()} | {error, Reason} when
         Reason :: {producer_not_found, topic()}
                 | {producer_not_found, topic(), partition()}
                 | {producer_down, noproc}
                 | client_down.
-find_producer(ClientId, Topic, Partition) ->
+find_producer(Client, Topic, Partition) ->
   try
-    SupPid = gen_server:call(ClientId, get_producers_sup_pid, infinity),
+    SupPid = gen_server:call(Client, get_producers_sup_pid, infinity),
     brod_producers_sup:find_producer(SupPid, Topic, Partition)
   catch exit : {noproc, _} ->
     {error, client_down}
   end.
 
+find_consumer(Client, Topic, Partition) ->
+  try
+    SupPid = gen_server:call(Client, get_consumers_sup_pid, infinity),
+    brod_consumers_sup:find_consumer(SupPid, Topic, Partition)
+  catch exit : {noproc, _} ->
+    {error, client_down}
+  end.
+
 %%%_* gen_server callbacks =====================================================
 
-init({ClientId, Endpoints, Config, Producers}) ->
+init({ClientId, Endpoints, Producers, Consumers, Config}) ->
   erlang:process_flag(trap_exit, true),
-  ets:new(?ETS(ClientId), [named_table, protected, {read_concurrency, true}]),
-  self() ! {init, Producers},
-  {ok, #state{ client_id = ClientId
-             , endpoints = Endpoints
-             , config    = Config
+  Tab = ets:new(?ETS(ClientId),
+                [named_table, protected, {read_concurrency, true}]),
+  self() ! {init, Producers, Consumers},
+  {ok, #state{ client_id     = ClientId
+             , endpoints     = Endpoints
+             , config        = Config
+             , producers_tab = Tab
+             }};
+init({Endpoints, Producers, Consumers, Config}) ->
```

Comment:
if producers are started here, get_producer will always return {error, client_down}.
part of the reason why i thought it's good to always force a unique client_id is to be able to use the name as the ets table, so the callers can get the producer pid from ets:lookup instead of going through a gen_server:call for each produce request.